pom.xml

```diff
@@ -44,6 +44,8 @@
     <modules>
         <module>modules/flink-ext</module>
         <module>modules/pub-sub</module>
+        <module>modules/spring-boot-autoconfigure</module>
+        <module>modules/spring-boot-client-autoconfigure</module>
```

Comment:
Can these 2 modules be combined and we can keep a single module `spring-boot-autoconfigure` or just `spring-boot` similar to spring module https://github.com/apache/ignite/blob/master/pom.xml#L56

Reply:
I think the answer is no, we can't.

The idea of the auto-configurer is to provide the user with zero-configuration integration with some third-party solution(Ignite in our case).

So, to start using `Ignite` bean inside the application the user has to add dependency `ignite-spring-boot-autoconfigure` to his project and that's all.

The same logic applies to the `ignite-client-spring-boot-autoconfigure`, with a minor difference - The user must provide `ignite-client.clientAddresses` property, otherwise the application would fail.

It seems, if we combine two autoconfiguration(`IgniteAutoConfiguration` and `IgniteClientAutoConfiguration`) in one dependency then init logic becomes inconsistent:

1. Should we start `Ignite` node, by default?
2. Should we fail if there are no `ignite-client.clientAddresses` in application properties?
3. If a user wants to have only `IgniteClient` in `BeanFactory` it will require additional configuration - disabling of `IgniteAutoConfiguration`.

My conclusion that we should keep two separate modules.

Reply:
Thank you @nizhikov for sharing the details, much appreciate it